lib/knife/changelog/changelog.rb

```diff
@@ -241,7 +241,13 @@ def generate_from_git_history(git, location, current_rev, rev_parse)
 
     def linkify(url, changelog)
       changelog.map do |line|
-        line.gsub(/^([a-f0-9]+) (.*)$/, '\2 (%s/commit/\1) ' % [url.chomp('.git')])
+        case url
+        when /gitlab/, /github/
+          line.gsub(/^([a-f0-9]+) (.*)$/, '\2 (%s/commit/\1) ' % [url.chomp('.git')])
+        else # assume gerrit style
+          hostname = "#{URI(url).scheme}://#{URI(url).hostname}"
+          line.gsub(/^([a-f0-9]+) (.*)$/, '\2 (%s/#/q/\1) ') % [hostname]
```

Comment:
I'm not fond this "gerrit by default" implementation, it assumes that any kind of url match gerrit, and that gerrit is setup at the root of the "vhost" (i.e. http://my-compa.ny/gerrit/group/project.git is not working).

Here is a suggestion:
```ruby
    GERRIT_REGEXP = %r{^(.*)/[^/]+/[^/]+(?:\.git)$}
    def linkify(url, changelog)
      format = case url
               when /gitlab/, /github/
                 "\\2 (#{url.chomp('.git')}/commit/\\1)"
               when GERRIT_REGEXP
                 "\\2 (#{::Regexp.last_match(1)}/#/q/\\1)"
               end

      format ? changelog.map { |line| line.sub(/^([a-f0-9]+) (.*)$/, format) } : changelog
    end
```

Reply:
seems reasonnable, even though the improvement is not that big.
I'll add a default case to at least have a clear exception when nothing is matched (or return a non-linkified version)

Reply:
the suggested solution return a non linkified version

Reply:
done